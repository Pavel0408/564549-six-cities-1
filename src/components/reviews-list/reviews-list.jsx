import React, {PureComponent} from "react";
import {Review} from "../review/review";
import PropTypes from "prop-types";
import {CommentForm} from "../comment-form/comment-form";
import {offerPropTypes} from "../../prop-types/offer-prop-types";
import {reviewPropTypes} from "../../prop-types/review-prop-type";
import {userPropTypes} from "../../prop-types/user-prop-types";

export class ReviewsList extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {reviewsError, reviews, user, sendReview, sendingError, isSending, activeOffer} = this.props;
    if (reviewsError && reviewsError.message) {
      return <h2>{reviewsError.message}</h2>;
    }

    return <section className="property__reviews reviews">
      <h2 className="reviews__title">Reviews Â· <span className="reviews__amount">{reviews.length}</span></h2>
      <ul className="reviews__list">
        {reviews.map((rewview) => {
          return <Review
            key={this.props.activeOffer.id.toString() + rewview.id}
            comment={rewview}
          />;
        })}
      </ul>
      {user && <CommentForm
        activeOffer={activeOffer}
        sendingError={sendingError}
        sendReview={sendReview}
        reviews={reviews}
        isSending={isSending}/>}
    </section>;
  }
}

ReviewsList.propTypes = {
  activeOffer: offerPropTypes,
  reviewsError: PropTypes.object,
  reviews: PropTypes.arrayOf(reviewPropTypes),
  fetchReviews: PropTypes.func,
  isAuthorizationRequired: PropTypes.bool,
  user: userPropTypes,
  sendReview: PropTypes.func,
  sendingError: PropTypes.object,
  isSending: PropTypes.bool
};
