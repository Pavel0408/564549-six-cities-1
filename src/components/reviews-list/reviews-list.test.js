import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";

import {ReviewsList} from "./reviews-list";

const dateFormat = new Intl.DateTimeFormat(`en-US`, {
  month: `long`,
  year: `numeric`
});


const mockComments = [
  {
    id: 1,
    userId: 3,
    userIsPro: true,
    userName: `Angelina`,
    userAvatar: `img/1.png`,
    rating: 4,
    message: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.,`,
    date: new Date(`2019-05-08T14:13:56.569Z`)
  }
];

const offersMock = {
  name: `Beautiful & luxurious apartment at great location`,
  image: `img/apartment-01.jpg`,
  price: 120,
  rating: 5,
  isPremium: false,
  isFavorite: true,
  coordinates: [52.3909553943508, 4.85309666406198],
  city: `Amsterdam`,
  id: 0
};

const user = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  avatar: `img/1.png`,
  isPro: false
};

describe(`testing ReviewList render`, () => {
  it(`ReviewList is render correctly`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <ReviewsList
            reviews={mockComments}
            activeOffer={offersMock}
            user={user}
            dateFormat={dateFormat}
          />
        </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
