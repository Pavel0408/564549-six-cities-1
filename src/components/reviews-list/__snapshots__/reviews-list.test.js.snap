// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ReviewList render ReviewList is render correctly 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height={54}
            src="img/1.png"
            width={54}
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Angelina
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.,
        </p>
        <time
          className="reviews__time"
          dateTime="2019-05-08T14:13:56.569Z"
        >
          May 2019
        </time>
      </div>
    </li>
  </ul>
  <form
    action="#"
    className="reviews__form form"
    method="post"
  >
    <label
      className="reviews__label form__label"
      htmlFor="review"
    >
      Your review
    </label>
    <div
      className="reviews__rating-form form__rating"
    >
      <input
        className="form__rating-input visually-hidden"
        defaultValue={5}
        id="5-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="5-stars"
        onClick={[Function]}
        title="perfect"
      >
        <svg
          className="form__star-image"
          height={33}
          width={37}
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        defaultValue={4}
        id="4-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="4-stars"
        onClick={[Function]}
        title="good"
      >
        <svg
          className="form__star-image"
          height={33}
          width={37}
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        defaultValue={3}
        id="3-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="3-stars"
        onClick={[Function]}
        title="not bad"
      >
        <svg
          className="form__star-image"
          height={33}
          width={37}
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        defaultValue={2}
        id="2-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="2-stars"
        onClick={[Function]}
        title="badly"
      >
        <svg
          className="form__star-image"
          height={33}
          width={37}
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        defaultValue={1}
        id="1-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="1-stars"
        onClick={[Function]}
        title="terribly"
      >
        <svg
          className="form__star-image"
          height={33}
          width={37}
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      className="reviews__textarea form__textarea"
      defaultValue=""
      id="review"
      name="review"
      onChange={[Function]}
      placeholder="Tell how was your stay, what you like and what can be improved"
    />
    <div
      className="reviews__button-wrapper"
    >
      <p
        className="reviews__help"
      >
        To submit review please make sure to set 
        <span
          className="reviews__star"
        >
          rating
        </span>
         and describe your stay with at least 
        <b
          className="reviews__text-amount"
        >
          50 characters
        </b>
        .
      </p>
      <input
        name="activeOfferId"
        type="hidden"
        value={0}
      />
      <button
        className="reviews__submit form__submit button"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</section>
`;
